src/sources/syslog.rs

```diff
@@ -291,6 +321,25 @@ pub fn udp(
             r#type = "udp"
         );
 
+        {
+            // SAFETY: We temporarily take ownership of the socket and return it by the end of this block scope.
+            let socket = unsafe { socket2::Socket::from_raw_fd(socket.as_raw_fd()) };
```

Comment:
Same for this.

Reply:
This code appears identical in all 3 places and like it's not dependent on the socket type. Could it be moved to a single common function?